parsers/test_zig/test_json.zig

```diff
@@ -0,0 +1,33 @@
+const std = @import("std");
+const allocator = std.testing.allocator;
+
+// /usr/local/zigDir/zig fmt .; /usr/local/zigDir/zig build-exe test_json.zig; ./test_json some_file.json; echo $?
+// echo '["Zig 0.8.0-dev.1354+081698156"]' > zig_only.json
+// python3 run_tests.py --filter=zig_only.json
+
+pub fn main() void {
+    const args = std.process.argsAlloc(allocator) catch |err| {
+        std.os.exit(1);
+    };
+
+    const file_name = args[1];
+    const file = std.fs.cwd().openFile(file_name, .{}) catch |err| {
+        std.os.exit(1);
+    };
+    const reader = file.reader();
+    const buf = reader.readAllAlloc(allocator, std.math.maxInt(usize)) catch |err| {
+        std.os.exit(1);
+    };
+
+    var p = std.json.Parser.init(allocator, false);
+    var tree = p.parse(buf) catch |err| {
```

Comment:
Avoid `std.json.Parser.parse` if you want control over json parsing, use `std.json.parse` instead.

Reply:
@daurnimator `std.json.parse` takes a `type` as its first arg. However for this repo, the type of json string that will be handed to our program is only known at runtime; https://github.com/nst/JSONTestSuite/tree/master/test_parsing